app/controllers/partners/approval_requests_controller.rb

```diff
@@ -7,7 +7,7 @@ def create
       if svc.errors.none?
         flash[:success] = "You have submitted your details for approval."
       else
-        flash[:error] = svc.errors.to_a.first
+        flash[:error] = "Please edit your profile to resolve these issues:  ".concat(svc.partner.profile.errors.full_messages.join(".  "))
```

Comment:
I guess the issue here would be if there is an error on some other aspect of the result we might not see it, or this might raise another error from being unable to crawl through.